bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/BulkMutation.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.grpc.async;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import com.google.bigtable.v1.MutateRowRequest;
+import com.google.bigtable.v1.MutateRowsRequest;
+import com.google.bigtable.v1.MutateRowsResponse;
+import com.google.common.util.concurrent.FutureCallback;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.SettableFuture;
+import com.google.protobuf.Empty;
+import com.google.rpc.Status;
+
+import io.grpc.StatusException;
+
+/**
+ * This class combines a collection of {@link MutateRowRequest}s into a single
+ * {@link MutateRowsRequest}. This class is not thread safe, and requires calling classes to make it
+ * thread safe.
+ */
+public class BulkMutation {
+  // TODO: make MAX_SIZE and MAX_COUNT configurable.
+  static final long MAX_SIZE = 1 << 21;
+  static final int MAX_COUNT = 50000;
+
+  private final List<SettableFuture<Empty>> futures = new ArrayList<>();
+  private final MutateRowsRequest.Builder builder;
+  private long size;
+
+  public BulkMutation(String tableName) {
+    this.builder = MutateRowsRequest.newBuilder().setTableName(tableName);
+    size = MutateRowsRequest.newBuilder().setTableName(tableName).build().getSerializedSize();
+  }
+
+  public synchronized SettableFuture<Empty> add(MutateRowRequest request) {
+    SettableFuture<Empty> future = SettableFuture.create();
+    futures.add(future);
+    MutateRowsRequest.Entry entry = MutateRowsRequest.Entry.newBuilder()
+      .setRowKey(request.getRowKey())
+      .addAllMutations(request.getMutationsList())
+      .build();
```

Comment:
Not sure how expensive the build() is but might want to do that work outside of a synchronized block.
Also the class comment indicates that the class is not thread safe but this method is synchronized.